### PR TITLE
AC: Added converter from YOLO labeling format to DetectionAnnotation

### PR DESCRIPTION
**Pull request by Anna-Mironova**

### CODE REVIEW

File: tools/accuracy_checker/accuracy_checker/annotation_converters/README.md [hunk 1]
```diff
@@ -502,6 +502,9 @@ The main difference between this converter and `super_resolution` in data organi
 * `vimeo90k` - converts Vimeo-90K dataset for a systematic evaluation of video processing algorithms to `SuperResolutionAnnotation`.
   * `annotation_file` - path to text file with list of dataset setuplets included in test.
   * `add_flow` - allows annotation of flow data (optional, default `False`).
+* `yolo_labeling` - converts object detection dataset with annotation in YOLO labeling format to `DetectionAnnotation`.
+  * `data_dir` - path to directory with annotation files in txt format and images.
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/README.md:506]: do images and labels are stored in the same directory or separated?

File: tools/accuracy_checker/accuracy_checker/annotation_converters/README.md [hunk 2]
```diff
@@ -502,6 +502,9 @@ The main difference between this converter and `super_resolution` in data organi
...
+  * `labels_file` - path to file with labels in txt format.
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/README.md:507]: should this parameters be optional?

File: tools/accuracy_checker/accuracy_checker/annotation_converters/yolo_labeling_converter.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+        meta = self.generate_meta()
+
+        return ConverterReturn(annotations, meta, content_errors)
+
+    def parse_annotation(self, annotation_file):
+        labels, x_mins, y_mins, x_maxs, y_maxs = [], [], [], [], []
+        for line in read_txt(self.data_dir / annotation_file):
+            label, x, y, width, height = line.split()
+            x_min, y_min, x_max, y_max = self.convert_from_yolo_to_bbox(float(x), float(y), float(width), float(height))
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/yolo_labeling_converter.py:67]: I think, this function can be useful for other tasks.
Possibly we can move it to utils and give more general name (e.g. xctr_yctr_w_h_to_x1y1x2y2)

File: tools/accuracy_checker/accuracy_checker/annotation_converters/yolo_labeling_converter.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        return np.array(labels), np.array(x_mins), np.array(y_mins), np.array(x_maxs), np.array(y_maxs)
+
+    @staticmethod
+    def convert_from_yolo_to_bbox(x, y, width, height):
+        x_min, y_min = (x - width / 2), (y - height / 2)
+        x_max, y_max = (x + width / 2), (y + height / 2)
+        return x_min, y_min, x_max, y_max
+
+    def generate_meta(self):
+        labels = read_txt(self.labels_file)
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/yolo_labeling_converter.py:82]: can we generate default labels (remember max label during annotation conversion and just map ranges)?

File: tools/accuracy_checker/accuracy_checker/annotation_converters/README.md (hunk 1 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/README.md:523]: yes, data_dir is directory for both images and annotations and they are stored in the same directory

File: tools/accuracy_checker/accuracy_checker/annotation_converters/README.md (hunk 2 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/README.md]: Done

File: tools/accuracy_checker/accuracy_checker/annotation_converters/yolo_labeling_converter.py (hunk 3 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/yolo_labeling_converter.py]: Done

File: tools/accuracy_checker/accuracy_checker/annotation_converters/yolo_labeling_converter.py (hunk 4 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/yolo_labeling_converter.py]: Added

**Review by eaidova — APPROVED** [review]

**Pull request merged**